app/frontend/src/cljs/teet/contract/contracts_view.cljs

```diff
@@ -47,67 +48,121 @@
                    (tr [:contracts :shortcuts option])]])]))
-          options)))
+          options))
+   [toggle-filters-visibility-button filters-visibility? toggle-filters-visibility]])
 
-(def search-fields
-  [[:road-number {:type TextField}]
-   [:project-name {:type TextField}]
-   [:contract-name {:type TextField}]
-   [:procurement-id {:type TextField}]
-   [:procurement-number {:type TextField}]
-   [:ta/region {:type select/select-enum
+(def filter-fields
+  [[:road-number {:type :search-field}]
+   [:project-name {:type :search-field}]
+   [:contract-name {:type :search-field}]
+   [:procurement-id {:type :search-field}]
+   [:procurement-number {:type :search-field}]
+   [:ta/region {:type :select-enum-field
                 :field-options {:attribute :ta/region
                                 :show-empty-selection? true}}]
-   [:contract-type {:type select/select-enum
+   [:contract-type {:type :select-enum-field
                     :field-options {:attribute :thk.contract/type
                                     :show-empty-selection? true}}]
-   [:contract-status {:type select/form-select
+   [:contract-status {:type :select-form-field
                       :field-options {:items contract-model/contract-statuses
                                       :format-item #(tr [:contract %])
                                       :attribute :thk.contract/status
-                                      :show-empty-selection? true}}]])
+                                      :show-empty-selection? true}}]
+   ])
 
-(defn search-inputs
-  [e! filter-values input-change search-fields]
-  [:div
-   (into [:div]
-         (mapv
-           (fn [[field-name {field-type :type
-                             field-options :field-options}]]
-             [field-type (merge {:e! e!
-                                 :label (tr [:contract :search field-name])
-                                 :value (field-name filter-values)
-                                 :start-icon icons/action-search
-                                 :on-change #(input-change
-                                               field-name
-                                               (if (= field-type TextField)
-                                                 (-> % .-target .-value)
-                                                 %))}
-                                field-options)])
-           search-fields))])
+(defmulti filter-input (fn [input-options filter-field-type]
+                         filter-field-type))
+
+(defmethod filter-input :search-field
+  [input-options]
+  [TextField (merge input-options
+               {:start-icon icons/action-search
+                :placeholder (tr [:search :quick-search])})])
+
+(defmethod filter-input :select-enum-field
+  [input-options]
+  [select/select-enum input-options])
+
+(defmethod filter-input :select-form-field
+  [input-options]
+  [select/form-select input-options])
```

Comment:
All of the defmethods should have the same amount of params as the defmulti definition.